examples/Example5_Formula_and_SMILES.txt

```diff
@@ -0,0 +1,16 @@
+# mzPAF annotation of "Levodopa_1" in 10.5281/zenodo.6325572
+1 89.0386	    25593   ?
+2 91.05419	39312   ?
+3 93.03366	93075	f{C6H5O}[M-H]/1.84ppm,s{OC=1C=CC=CC1}-H/1.84ppm

```

Comment:
Instead of:
f{C6H5O}[M-H]/1.84ppm,s{OC=1C=CC=CC1}-H/1.84ppm
do:
f{C6H5O}[M-H]/1.84ppm,s{OC=1C=CC=CC1}[M-H]/1.84ppm
?
formula policy: include all nuclei, including charged nuclei
smiles policy: just describe the neutral form, and denote the charge with [M+xx] notation

For a radical ion:
s{OC=1C=CC=CC1}[M-e]
or
s{OC=1C=CC=CC1}[M+]

Reply:
All, I wrote something to match what we discussed for the mzPAF specificiation doc. This goes under the Section "SMILES strings for chemical compounds." Please help review (don't want to use a separate pull request):

Chemical compounds may also be expressed using SMILES notation (https://en.wikipedia.org/wiki/Simplified_molecular-input_line-entry_system). SMILES strings MUST be prefixed by ‘s’ and enclosed in curly braces. 

For example: 

s{CN=C=O}/-0.55ppm
s{COc(c1)cccc1C#N}^2/1.29ppm 

Since the structure of the adduct ion is often unclear, the SMILES string MUST NOT include the charge-bearing moieties (H+, Na+, etc.) which may be attached at unspecified sites. Therefore, the SMILES string MUST correspond to the neutral molecule, and the charge-bearing moieties should be specified by the adduct notation, except when only the addition of protons are involved. In the example above s{CN=C=O} should be interpreted as the neutral molecule methyl isocyanate H3C-N=C=O with an additional proton attached to it at an unspecified location, for a +1 charge (a charge state of +1 is assumed for all peak annotations unless specified with a ^c suffix). This also means that the mass of the ion should be calculated from the neutral mass of methyl isocyanate plus that of a proton. 

To indicate a non-protonated adduct ion, one MUST use the [M+X] notation. For example, s{COc(c1)cccc1C#N}[M+H+Na]^2 denotes an adduct ion of 2+ charge state, in which the neutral molecule of COc(c1)cccc1C#N  (3-methoxybenzonitrile) is attached to one proton and one Na+ ion at some unspecified sites. 

Note that while the SMILES format allows for the specification of charged nuclei, it only supports the case where the molecular structure is fully known. Although sometimes it is possible to place the charge-bearing moiety at an exact location with high confidence, in order to avoid confusion about how the mass should be calculated, the SMILES string MUST encode neutral molecules only. The charge state and the charge-bearing moiety(-ies) (except protons) MUST be encoded as suffixes following the SMILES string, as described elsewhere in this document.  

Likewise, although SMILES allows the specification of isotopes, e.g. [13C] for carbon-13, one MUST not include such heavy isotopes in the SMILES string, as it is in the context of peak annotation, it is impossible to pinpoint the exact location of the heavy atom(s) based solely on the measured m/z value. All peak annotations employing SMILES string in this format are assumed to be monoisotopic. Higher isotopic ions MUST be specified with the +ni suffix as described elsewhere in this document.    

Reply:
I believe that last week we agreed that the molecular formula would contain the proton, thus for the SMILES to have an implicit proton feels slightly inconsistent. I feel that [M+H] should always be included next to the SMILES. Otherwise I very much like Henry's text.

Reply:
I agree that it feels slightly inconsistent, but yet seems like the best choice. I agree that it seems clearer to require the [M+H] and would support that, but don't feel strongly.

Reply:
Agree. SMILES will probably be used more often by metabolomics people, who are familiar with different adducts, and thus the explicit requirement is not overly cumbersome. Otherwise great text Henry.

Reply:
So the examples would change from:
s{CN=C=O}/-0.55ppm
s{COc(c1)cccc1C#N}^2/1.29ppm 

to

s{CN=C=O}[M+H]/-0.55ppm
s{COc(c1)cccc1C#N}^2[M+2H]/1.29ppm 

Also add an example for loss of electron?
s{CN=C=O}[M-e]/-0.55ppm

Loss of electron and addition of proton?  [M+H-e]

Clarify that H means a proton
State the [M+X] describes added or subtracted nuclei (or electrons) that confer the charge to the whole ion

[M+H]
[M+Na]

Henry will update the draft based on this and we will re-look.

Reply:
New attempt:

Chemical compounds may also be expressed using SMILES notation (https://en.wikipedia.org/wiki/Simplified_molecular-input_line-entry_system). SMILES strings MUST be prefixed by ‘s’ and enclosed in curly braces.

For example: 

s{CN=C=O}[M+H]/-0.55ppm
s{COc(c1)cccc1C#N}^2[M+H+Na]/1.29ppm 

Since the structure of the adduct ion is often unclear, the SMILES string MUST NOT include the charge-bearing moieties (H+, Na+, etc.) which may be attached at unspecified sites. Therefore, the SMILES string MUST correspond to the neutral molecule, and the charge-bearing moieties should be specified by the adduct notation as described in Section 4.8. 

In the example above s{CN=C=O}[M+H] should be interpreted as the neutral molecule methyl isocyanate H3C-N=C=O with an additional proton attached to it at an unspecified location, to yield a 1+ charge (a charge state of 1+ is assumed for all peak annotations unless specified with a ^c suffix). This also means that the mass of the ion should be calculated from the neutral mass of methyl isocyanate plus that of a proton. Likewise, s{COc(c1)cccc1C#N}^2[M+H+Na] denotes an adduct ion of 2+ charge state, in which the neutral molecule of COc(c1)cccc1C#N  (3-methoxybenzonitrile) is attached to one proton and one Na+ ion at some unspecified sites. 

Note that while the SMILES format allows for the specification of charged nuclei, it only supports the case where the molecular structure is fully known. Although sometimes it is possible to place the charge-bearing moiety at an exact location with high confidence, in order to avoid confusion about how the mass should be calculated, the SMILES string MUST encode neutral molecules only. The charge state and the charge-bearing moiety(-ies) MUST be encoded in the adduct notation following the SMILES string, as described in Section 4.8 this document. In addition, if SMILES strings are used, the [M+nH] suffix MUST always be included even if the adduct is only protonated (unlike for peptides for which protonated adducts are assumed if omitted). 

For the purpose of mass calculation, it is assumed that what follows the M in the adduct notation are ions which should confer the correct charge state, i.e., [M+Na] means the addition of a 1+ sodium ion, not a sodium atom, whereas [M+HCOO] means the addition of formate HCOO-, not the neutral radical of HCOO. It should be noted that sometimes the adduct notation is used in the literature to also specify neutral loss/gain to the neutral analyte, in addition to the charge-bearing moeity(-ies). Since this format already provides for a mechanism to specify neutral loss/gain (Section 4.5), it is recommended that what follows the M in the adduct notation SHOULD only include charge-bearing ions, and neutral loss/gain SHOULD NOT be included in the adduct notation. For example,  s{OCCCC=OOH}-H2O[M+H] is preferred over s{OCCCC=OOH}[M-H2O+H] to denote 4-hydroxybutyric acid with a water loss (with exact structure unknown), charged with an additional proton.

If the ion is generated by the gain/loss of electrons only, then the adduct notations of [M+/-ne] MUST be used. For example, [M-e] stands for the 1+ ion generated when the neutral molecule specified by the SMILES string loses an electron. [M+2e] stands for the 2- ion generated when two electrons are added to the neutral molecule. The shorthand of [M+] to mean [M-e] is not allowed.

Likewise, although SMILES allows the specification of isotopes, e.g. [13C] for carbon-13, one MUST not include such heavy isotopes in the SMILES string, as it is in the context of peak annotation, it is impossible to pinpoint the exact location of the heavy atom(s) based solely on the measured m/z value. All peak annotations employing SMILES string in this format are assumed to be monoisotopic. Higher isotopic ions MUST be specified with the +ni suffix as described in Section 4.6.

Reply:
After discussion, we will remove the last paragraph as not necessary.